kolibri/core/content/management/commands/generate_schema.py

```diff
@@ -10,6 +11,51 @@
 from django.core.management import call_command
 from django.core.management.base import BaseCommand
 from django.db import connections
+
+# Compatibility layer for Python 3.12+ where ArgSpec is removed
+if not hasattr(inspect, "ArgSpec"):
+
+    class ArgSpec:
+        def __init__(self, args, varargs, keywords, defaults):
+            self.args = args
+            self.varargs = varargs
+            self.keywords = keywords
+            self.defaults = defaults
+
+    def getargspec(func):
+        spec = inspect.getfullargspec(func)
+        return ArgSpec(
+            args=spec.args,
+            varargs=spec.varargs,
+            keywords=spec.varkw,
+            defaults=spec.defaults,
+        )
+
+    inspect.ArgSpec = ArgSpec
+    inspect.getargspec = getargspec
+
+# SQLAlchemy 2.0 compatibility patch
```

Comment:
Why do we need a compatibility patch with SQLAlchemy 2? We explicitly pin to less than 2, and I've not had any issues installing SQLAlchemy 1.4.49 in Python 3.12.

Reply:
fixed, somehow I mixed my virtual environments